MiscVBAFunctions/Modules/MiscTables.bas

```diff
@@ -191,3 +191,142 @@ Function TableColumnToArray(TableDicts As Collection, ColumnName As String) As V
+    ' If NumRows is set to "0", the table will instead be resized to "1" and the
+    ' content in that 1st row will be cleared.
+    '
+    ' Args:
+    '   SelectedLO: Selected List Object.
+    '   NumRows: Number of desired rows.
+    
+    Dim oldNumRows As Long
+    oldNumRows = SelectedLO.ListRows.Count
+    
+    ' Don't allow tables to be resized to zero rows.  Excel won't really do
+    ' this anyway - it's possible to get the DataBodyRange to be empty, in
+    ' which case the single row displayed in the table is the Insert row, but
+    ' this creates more problems than it solves.  Instead, resize the table to
+    ' one row, and set a flag to clear out any remaining data later.
+    Dim clearTable As Boolean
+    clearTable = False
+    If NumRows = 0 Then
+        NumRows = 1
+        clearTable = True
+    End If
+    
+    ' Resize the table (add 1 to the number of rows because mListObject.Range
+    ' includes the header row).
+    SelectedLO.Resize _
+        SelectedLO.Range.Resize( _
+            NumRows + 1, _
+            SelectedLO.ListColumns.Count)
+    
+    ' If the table is resized to have one row, but the row contains no data,
+    ' the row will be treated as the Insert row, and the data row count will
+    ' remain zero.  This will cause problems since the table doesn't actually
+    ' have a DataBodyRange.  To avoid this situation, put a space in the first
+    ' column, which will cause the Insert row to change to a data row.  After
+    ' setting the value once, it can be removed and the row will remain part
+    ' of the DataBodyRange.
+    
+    If NumRows = 1 And SelectedLO.ListRows.Count = 0 Then
+        SelectedLO.Range.Cells(1, 1).Offset(1, 0).Value = " "
+        SelectedLO.DataBodyRange.ClearContents
+    End If
+    
+    ' The user requested that the table be resized to zero rows.  We resized
+    ' it down to one row, now clear out the data.
+    
+    If clearTable Then
+        SelectedLO.DataBodyRange.ClearContents
+        'ClearSort
+    End If
+    
+    ' If the new number of rows is less than the old number of rows, clear out
+    ' the rows that were just removed from the table.
+    If NumRows < oldNumRows Then
+        SelectedLO.DataBodyRange _
+            .Offset(NumRows, 0) _
+            .Resize(oldNumRows - NumRows, SelectedLO.ListColumns.Count) _
+            .ClearContents
+    End If
+End Sub
+
+
+Public Function GetColumnLO(SelectedLO As ListObject, ColumnName As String) As Range
+    ' Returns the data range for the given column of this Excel table.
+    ' An error will be raised if the selected Column name doesn't exist in the given List Object.
+    '
+    ' Args:
+    '   SelectedLO: List Object to fetch the column from
+    '   ColumnName: Name of the column where the data will be fetched from.
+    '
+    ' Returns:
+    '   Data Range of the given column.
+    
+    Dim listCol As ListColumn
+    On Error GoTo noDataRange
```

Comment:
I think using on Error here could `obfuscate` the actual issue/error when using the function.
We should try use `On Error GoTo` sparingly and rather raise errors more specifically.

For example:
```
If Not HasKey(SelectedLO.ListColumns, ColumnName) Then
    ' Raise error that column name doesn't exist
End If

If SelectedLO.DataBodyRange is Nothing Then
    ' I was thinking this should raise an error, but actually, it should just return `Nothing`, as the DataBodyRange is nothing
    ' So we can just exit the function and it will return Nothing as there is no data in the list object
    ' This is another good reason for why a ListObject with 0 rows should have a DataBodyRange of Nothing. Else this function will return a 1x1 range with value VbNullString, which is not actually the case! There is no data
    Exit Function 
' Raise error "Cannot select
End If
```